### ISSUE TITLE
Replace `<a />` with `<Link />` in markdown fields

### ISSUE THREAD

**Issue by seancdavis** [ample/gatsby-starter-ample#121]
What I'm interested in here is having links within markdown frontmatter fields as well as the main body content use the `<Link />` component. It's _possible_ this is already happening in one of those cases, but it's worth exploring.

**Comment by seancdavis** [comment]
I wonder if we could take a similar approach as we did in #167 when tapping into link nodes in markdown frontmatter fields.

**Comment by seancdavis** [comment]
I did quite a bit of digging with this, and as it turns out we can't make use of #167, at least not easily. Because gatsby-link is a react component, we'd have to switch to using MDX.

There _may_ be some benefit in using MDX, but @dholloran-ample is exploring that as part of #194.

After going down a rabbit hole, I found that a plugin we're using, [gatsby-plugin-catch-links](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-catch-links) works on _all_ links on a page. It won't prefetch the content, like gatsby-link, but it keeps the routing within the application.

Therefore, I'm closing this, as we effectively already have the behavior this issue was looking to achieve.
